core/src/main/java/org/apache/carbondata/core/indexstore/PartitionSpec.java

```diff
@@ -43,8 +43,6 @@
    */
   private List<String> partitions;
 
-  private transient Path locationPath;
```

Comment:
This doesn't seem like the actual root cause of the issue, as the exception trace shows HiveExternalCatalog as the Non-serializable field. Please check how HiveExternalCatalog is being sent for serialization

Reply:
changed